documentation/debugging.rst

```diff
@@ -0,0 +1,106 @@
+Debugging and troubleshooting 
+##########
+
+This section is dedicated to the description of problems that 
+might appear while creating the python bindings with binder and the ways to avoid them.
+
+Below are some helpful tips that might help to make the bindings.
+
+
+---------------
+Inconsistencies
+---------------
+
+Binder moves down the ``all_includes_file`` file sequentially, sometimes ending up with errors.  
+This is almost always caused by the ``all_includes_file`` being
+different between runs.  The order shouldn't be important, but nail it down to at least be
+consistent, and then move on to the next step.
+
+--------------
+Build failures
+--------------
+
+Even when the bindings were generated successfully,  there might be compilation errors when building the modules from the generated sources.
+Quite often  the errors are caused by the implementation of the С++standard library, when the headers of the standard library 
+include each other, or include implementation-specific headers. 
+Many cases like that are already handled in the functions from the ``source/types.cpp`` file, 
+using the knowledge of the existing STL implementations.
+However some cases might still be missing, e.g. for the newest or not wide-spread versions of STL. 
+An example of debugging for these cases is described below.
+
+
+On systems with GNU STL, the compilation errors for the cases not handled by the ``source/types.cpp``,
+would manifest itself with an abundance of errors complaining about some headers from the ``bits`` directory, which typically 
+contains the details of particular STL implementation.
+
+For instance, the compilation could fail with an error message:
+
+.. code-block:: console
+
+    FAILED: CMakeFiles/statvec.dir/std/complex.o 
+    In file included from std/complex.cpp:1:0:
+    /usr/include/c++/7/bits/stl_construct.h: In function ‘void std::_Destroy(_ForwardIterator, _ForwardIterator)’:
+    **bunch of garbage below here**
```

Comment:
Could you please replace this with a more technical (and correct) wording? Something like "long and cryptic error message" seems more appropriate to me.

Reply:
OK. Done.

Reply:
hm... could it be that changes have not been pushed? Because i still see original wording... @andriish could you please double check on that? Thanks,

Reply:
never mind i will just edit the files from browser

Reply:
Hi @lyskov,

yes, it looks one part of the commit was lost.

Thanks for fixing it

Best regards,
Andrii